crates/polars-expr/src/reduce/bitwise.rs

```diff
@@ -0,0 +1,254 @@
+        ca.or_reduce()
+    }
+}
+
+impl<T> NumericReduction for BitwiseXor<T>
+where
+    T: PolarsNumericType,
+    T::Native: BitXor<Output = T::Native> + Zero,
+    T::Native: NativeType,
+    PrimitiveArray<T::Native>: BitwiseKernel<Scalar = T::Native>,
+{
+    type Dtype = T;
+
+    #[inline(always)]
+    fn init() -> T::Native {
+        T::Native::zero() // all_zeroes(), the identity element
+    }
+
+    #[inline(always)]
+    fn combine(a: T::Native, b: T::Native) -> T::Native {
+        a ^ b
+    }
+
+    #[inline(always)]
+    fn reduce_ca(ca: &ChunkedArray<Self::Dtype>) -> Option<T::Native> {
+        ca.xor_reduce()
+    }
+}
+
+#[derive(Default)]
+pub struct BoolXorGroupedReduction {
+    values: MutableBitmap,
+    mask: MutableBitmap,
+    evicted_values: BitmapBuilder,
+    evicted_mask: BitmapBuilder,
+}
+
+impl GroupedReduction for BoolXorGroupedReduction {
+    fn new_empty(&self) -> Box<dyn GroupedReduction> {
+        Box::new(Self::default())
+    }
+
+    fn reserve(&mut self, additional: usize) {
+        self.values.reserve(additional);
+        self.mask.reserve(additional)
+    }
+
+    fn resize(&mut self, num_groups: IdxSize) {
+        self.values.resize(num_groups as usize, false);
+        self.mask.resize(num_groups as usize, false);
+    }
+
+    fn update_group(
+        &mut self,
+        values: &Column,
+        group_idx: IdxSize,
+        _seq_id: u64,
+    ) -> PolarsResult<()> {
+        assert!(values.dtype() == &DataType::Boolean);
+        let values = values.as_materialized_series_maintain_scalar();
+        let ca: &BooleanChunked = values.as_ref().as_ref();
+        if let Some(value) = ca.xor_reduce() {
+            // SAFETY: indices are in-bounds guaranteed by trait
+            unsafe {
+                self.values.xor_pos_unchecked(group_idx as usize, value);
+            }
+        }
+        if ca.len() != ca.null_count() {
+            self.mask.set(group_idx as usize, true);
+        }
+        Ok(())
+    }
+
+    unsafe fn update_groups_while_evicting(
+        &mut self,
+        values: &Column,
+        subset: &[IdxSize],
+        group_idxs: &[EvictIdx],
+        _seq_id: u64,
+    ) -> PolarsResult<()> {
+        assert!(values.dtype() == &DataType::Boolean);
+        assert!(subset.len() == group_idxs.len());
+        let values = values.as_materialized_series(); // @scalar-opt
+        let ca: &BooleanChunked = values.as_ref().as_ref();
+        let arr = ca.downcast_as_array();
+        unsafe {
+            // SAFETY: indices are in-bounds guaranteed by trait.
+            for (i, g) in subset.iter().zip(group_idxs) {
+                let ov = arr.get_unchecked(*i as usize);
+                if g.should_evict() {
+                    self.evicted_values.push(self.values.get_unchecked(g.idx()));
+                    self.evicted_mask.push(self.mask.get_unchecked(g.idx()));
+                    self.values.xor_pos_unchecked(g.idx(), ov.unwrap_or(false));
```

Comment:
This one should be `set_unchecked`, not `xor_pos_unchecked`. We evicted the old value so we should initialize with a fresh value instead of continuing to update it.